src/gui/DatabaseWidget.cpp

```diff
@@ -1520,13 +1524,34 @@ void DatabaseWidget::onGroupChanged()
 
 void DatabaseWidget::onDatabaseModified()
 {
-    if (!m_blockAutoSave && config()->get(Config::AutoSaveAfterEveryChange).toBool()) {
+    refreshSearch();
+    int autosaveDelay = m_db->metadata()->autosaveDelay() * 60 * 1000;  // min to msec for QTimer
+    bool autosaveAfterEveryChangeConfig = config()->get(Config::AutoSaveAfterEveryChange).toBool();
+    if (m_autosaveDelayTimer->isActive()) {
```

Comment:
Looks like these 2 conditions have the same body and can be simplified:
```c++
    if (m_autosaveDelayTimer->isActive() ||
        (autosaveDelay > 0 && autosaveAfterEveryChangeConfig)) {
        m_autosaveDelayTimer->start(autosaveDelay);
        return;
    }
```

Reply:
Fixed at [abfb352](https://github.com/keepassxreboot/keepassxc/pull/9100/commits/abfb352eaa2fc888dc33d7756c3cd287e9fba2a8)
Looks good